routers/api/v1/api.go

```diff
@@ -975,6 +981,53 @@ func Routes() *web.Route {
 				}, reqAnyRepoReader())
 				m.Get("/issue_templates", context.ReferencesGitRepo(false), repo.GetIssueTemplates)
 				m.Get("/languages", reqRepoReader(models.UnitTypeCode), repo.GetLanguages)
+				m.Group("/packages", func() {
+					m.Group("/generic", func() {
+						m.Group("/{packagename}/{packageversion}/{filename}", func() {
+							m.Get("", generic.DownloadPackageFile)
```

Comment:
All downloads and read routers needs middleware
`reqRepoReader(models.UnitTypePackages),`